codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/ImportDeclarations.java

```diff
@@ -0,0 +1,67 @@
+package software.amazon.polymorph.smithygo;
+
+import software.amazon.smithy.codegen.core.CodegenException;
+import software.amazon.smithy.codegen.core.ImportContainer;
+import software.amazon.smithy.codegen.core.Symbol;
+import software.amazon.smithy.utils.StringUtils;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+public class ImportDeclarations implements ImportContainer {
+
+    private final Map<String, String> imports = new TreeMap<>();
+
+    ImportDeclarations addImport(String importPath, String alias) {
+        String importAlias = CodegenUtils.getDefaultPackageImportName(importPath);
+        if (!StringUtils.isBlank(alias)) {
+            if (alias.equals(".")) {
+                // Global imports are generally a bad practice.
+                throw new CodegenException("Globally importing packages is forbidden: " + importPath);
+            }
+            importAlias = alias;
+        }
+//        // Ensure that multiple packages cannot be imported with the same name.
+//        if (imports.containsKey(importAlias) && !imports.get(importAlias).equals(importPath)) {
+//            throw new CodegenException("Import name collision: " + importAlias
+//                                               + ". Previous: " + imports.get(importAlias) + "New: " + importPath);
+//        }
```

Comment:
Question:
Did we comment this out or did Smithy-Go's author's?
If it was us, why?

Reply:
Same as above.